src/common/collection.c

```diff
@@ -831,6 +831,74 @@ void dt_collection_split_operator_datetime(const gchar *input, char **number1, c
   g_regex_unref(regex);
 }
 
+void dt_collection_split_operator_exposure(const gchar *input, char **number1, char **number2, char **operator)
+{
+  GRegex *regex;
+  GMatchInfo *match_info;
+  int match_count;
+
+  *number1 = *number2 = *operator= NULL;
+
+  // we test the range expression first
+  regex = g_regex_new("^\\s*\\[\\s*(1/)?([0-9]+\\.?[0-9]*)(\")?\\s*;\\s*(1/)?([0-9]+\\.?[0-9]*)(\")?\\s*\\]\\s*$", 0, 0, NULL);
+  g_regex_match_full(regex, input, -1, 0, 0, &match_info, NULL);
+  match_count = g_match_info_get_match_count(match_info);
+
+  if(match_count == 6 || match_count == 7)
+  {
+    *number1 = g_match_info_fetch(match_info, 2);
+    if(strstr(g_match_info_fetch(match_info, 1), "1/") != NULL)
+    {
+      float n1 = atof(*number1);
```

Comment:
the matching issue is probably because you are converting to float here (in C) and later you just convert back to float (in SQL) for the matching... No surprise you have rounding issues, and so matching issues.